lib/trino-collect/src/main/java/io/trino/collect/cache/EvictableCacheBuilder.java

```diff
@@ -126,19 +127,23 @@ public EvictableCacheBuilder<K, V> recordStats()
      */
     public EvictableCacheBuilder<K, V> shareResultsAndFailuresEvenIfDisabled()
     {
-        checkState(!disabledCacheImplementation.isPresent(), "disabledCacheImplementation already set");
-        disabledCacheImplementation = Optional.of(DisabledCacheImplementation.GUAVA);
-        return this;
+        return disabledCacheImplementation(DisabledCacheImplementation.GUAVA);
     }
 
     /**
      * Choose a behavior for case when caching is disabled that prevents data and failure sharing between concurrent callers.
      * Note: disabled cache won't report any statistics.
      */
     public EvictableCacheBuilder<K, V> shareNothingWhenDisabled()
+    {
+        return disabledCacheImplementation(DisabledCacheImplementation.NOOP);
+    }
+
+    @VisibleForTesting
+    EvictableCacheBuilder<K, V> disabledCacheImplementation(DisabledCacheImplementation cacheImplementation)
```

Comment:
consider name it as setter `setDisabledCacheImplementation`

Reply:
the other builder methods in this class don't have `set` prefix